### PR TITLE
[ Xedra Evolved ] Add some additional missions to gain best friend status with an npc.

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Mods "XE Missions for best friend quest"

#### Purpose of change
Adds some new missions in on the quest for a vampire to gain a best friend in a lonely world.

#### Describe the solution
This adds some missions that have to be done after getting an NPCs trust up in order to become their best friend.

#### Describe alternatives you've considered
More or less missions.  Different missions.  More random options

### CODE REVIEW

**Comment by SariusSkelrets** [comment]
Thanks for adding this. It'll make the process more involved than just doing a generic NPC mission then spamming chitchat until the values are high enough

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 1]
```diff
@@ -98,5 +118,219 @@
...
+        { "case": 1, "effect": { "assign_mission": "mission_best_friend_gamehunt1" } },
+        { "case": 2, "effect": { "assign_mission": "mission_best_friend_gamehunt2" } },
+        { "case": 3, "effect": { "assign_mission": "mission_best_friend_gamehunt3" } }
+      ]
+    }
+  },
+  {
+    "id": "mission_best_friend_gamehunt1",
+    "type": "mission_definition",
+    "name": { "str": "Help me find my favorite game.  Pandemic" },
```
[the remark this answers is not in the document]
Comment (Maleclypse) [reply data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:237]: Good catch I forgot some have changed names and some have the same as the id.

Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:237]:
```suggestion
    "name": { "str": "Help me find my favorite game.  Pathogen" },
```

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 2]
```diff
@@ -98,5 +118,219 @@
...
+    "description": "Could you help me find my favorite game?  It's a board game called Pandemic.  At least the diseases in this game can be beaten.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:238]:
```suggestion
    "description": "Could you help me find my favorite game?  It's a board game called Pathogen.  At least the diseases in this game can be beaten.",
```

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 3]
```diff
@@ -98,5 +118,219 @@
...
+          "decay_start": "60 minutes"
+        },
+        { "u_add_var": "best_friend", "value": "yes" }
+      ]
+    }
+  },
+  {
+    "id": "mission_best_friend_gamehunt2",
+    "type": "mission_definition",
+    "name": { "str": "Help me find my favorite game.  Scythe" },
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:271]:
```suggestion
    "name": { "str": "Help me find my favorite game.  Steam Gears" },
```

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 4]
```diff
@@ -98,5 +118,219 @@
...
+    "description": "Could you help me find my favorite game?  It's a board game called Scythe.  Alternate history is so much better than real history.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:272]:
```suggestion
    "description": "Could you help me find my favorite game?  It's a board game called Steam Gears.  Alternate history is so much better than real history.",
```

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 5]
```diff
@@ -98,5 +122,219 @@
...
+    }
+  },
+  {
+    "id": "mission_best_friend_bookhunt2",
+    "type": "mission_definition",
+    "name": { "str": "Help me find my favorite book.  Last and First Men." },
+    "description": "Could you help me find my favorite book?  It's Last and First Men by Olaf Stapledon.  It really spoke to me before the end and even more now.",
+    "goal": "MGOAL_FIND_ITEM",
+    "difficulty": 2,
+    "item": "scifi1_46",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:166]:
```suggestion
    "name": { "str": "Help me find my favorite book.  Brave New World." },
    "description": "Could you help me find my favorite book?  It's Brave New World by Aldous Huxley.  It really spoke to me before the end and even more now.",
    "goal": "MGOAL_FIND_ITEM",
    "difficulty": 2,
    "item": "book_fict_soft_scifi_brave",
```

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 6]
```diff
@@ -98,5 +122,219 @@
...
+    }
+  },
+  {
+    "id": "mission_best_friend_bookhunt3",
+    "type": "mission_definition",
+    "name": { "str": "Help me find my favorite book.  A Scanner Darkly." },
+    "description": "Could you help me find my favorite book?  It's A Scanner Darkly, by Phillip K. Dick.  I just love that story and it makes me think of better days before everyone died.",
+    "goal": "MGOAL_FIND_ITEM",
+    "difficulty": 2,
+    "item": "scifi1_17",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:199]:
```suggestion
    "item": "book_fict_soft_docu_AWT",
```

File: data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json [hunk 7]
```diff
@@ -98,5 +122,219 @@
...
+          "decay_start": "60 minutes"
+        }
+      ]
+    }
+  },
+  {
+    "id": "mission_best_friend_bookhunt3",
+    "type": "mission_definition",
+    "name": { "str": "Help me find my favorite book.  A Scanner Darkly." },
+    "description": "Could you help me find my favorite book?  It's A Scanner Darkly, by Phillip K. Dick.  I just love that story and it makes me think of better days before everyone died.",
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/npc/npc_vampiresacrifice_chat.json:196]:
```suggestion
    "name": { "str": "Help me find my favorite book.  All Things Weird." },
    "description": "Could you help me find my favorite book?  It's All Things Weird, by Felicity Powell.  It gives a great sense of humor to the end of the world!",
```

**Pull request merged**